eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Plutus/Evaluate.hs

```diff
@@ -209,17 +211,17 @@ merge f (x : xs) zs = merge f xs (gg x zs)
 
 -- | Evaluate a list of Plutus scripts. All scripts in the list must evaluate to `True`.
 evalPlutusScripts ::
-  EraTx era =>
+  (EraTx era, Crypto c) =>
   Tx era ->
-  [PlutusWithContext] ->
-  ScriptResult
+  [PlutusWithContext c] ->
+  ScriptResult c
```

Comment:
`c` and `era` should never appear together, since `EraCrypto era` is always the crypto for the era 
```suggestion
  [PlutusWithContext (EraCrypto era)] ->
  ScriptResult (EraCrypto era)
```

Reply:
Ah yes, I was planning to change those since at one point I realised that there is no point in introducing an extra type variable where we already have the `era`, it just seemed redundant. Hence why at certain parts I only used `EraCrypto era` and was about to clean up the rest. But I didn't know that having both is actually discouraged for other reasons than redundancy. Thanks!

By the way, what necessitated the introduction of `EraCrypto` in the first place (given the fact that `Crypto c` already seems extraneous at this point since it is always `Crypto StandardCrypto`)?

Reply:
Not sure if this is what you're asking, but: if you have an `Era era` constraint (or a super class, eg: `EraTx` in this case),  then you have already the `Crypto` type for it, and you reference it with `EraCrypto era`   (https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/src/Cardano/Ledger/Conway/Era.hs#L42)

Reply:
Oh or maybe you're asking why was it necessary to be introduced at all?  Nevermind my comment if so :)

Reply:
The latter. :) I was wondering why it was introduced at all. I mean sure, it is more descriptive than the `Crypto c` constraint but still, I was wondering if there were some other reasons.
Regardless, thanks for the clarification regarding the `Era` constraints as well. :)

Reply:
> I was wondering why it was introduced at all.

We were discussing just that on Wednesday meeting. :smile: 

I suspect there was two reasons:
* Ability to use different crypto algortihms  during testing. Which is sort of helpful for VRF, but not at all for hashes and signing algorithms
* Assumption that it would be more versatile, which proved to be incorrect.

In other words,  this is the decision that was made early on and now we are stuck with it. But, maybe we'll be able to get rid of it in the future. I have a descent plan on how it can be made possible.

Reply:
> I suspect there was two reasons:
> 
>     * Ability to use different crypto algortihms  during testing. Which is sort of helpful for VRF, but not at all for hashes and signing algorithms
> 
>     * Assumption that it would be more versatile, which proved to be incorrect.

But the above is about the `Crypto` constraint, right? I remember we were discussing that. I was wondering about the history behind `EraCrypto`. Was it introduced for convenience?